includes/Context.php

```diff
@@ -384,15 +364,40 @@ private function create_entity_for_post( \WP_Post $post ) {
 	 *
 	 * @since n.e.x.t
 	 *
-	 * @param string $url Input URL.
-	 * @return string URL that starts with the site reference URL.
+	 * @param string $url Optional. Input URL. If not provided, returns the plain reference site URL.
+	 * @return string URL that starts with the reference site URL.
 	 */
-	private function filter_reference_url( $url ) {
-		$reference_site_url = $this->get_reference_site_url();
-		$orig_site_url      = home_url();
+	private function filter_reference_url( $url = '' ) {
+		$orig_site_url = untrailingslashit( home_url() );
+		$site_url      = $orig_site_url;
+
+		/**
+		 * Filters the reference site URL to use for stats.
+		 *
+		 * This can be used to override the current site URL, for example when using the plugin on a non-public site,
+		 * such as in a staging environment.
+		 *
+		 * @since 1.0.0
+		 *
+		 * @param string $site_url Reference site URL, typically the WordPress home URL.
+		 */
+		$site_url = apply_filters( 'googlesitekit_site_url', $site_url );
+
+		// Ensure this is not empty.
+		if ( empty( $site_url ) ) {
+			$site_url = $orig_site_url;
+		} else {
+			$site_url = untrailingslashit( $site_url );
+		}
+
+		// If no URL given, just return the reference site URL.
+		if ( empty( $url ) ) {
+			return $site_url;
+		}
 
-		if ( $orig_site_url !== $reference_site_url ) {
-			$url = str_replace( $orig_site_url, $reference_site_url, $url );
+		// Replace original site URL with the reference site URL.
+		if ( $orig_site_url !== $site_url ) {
+			$url = str_replace( $orig_site_url, $site_url, $url );
```

Comment:
I found this a bit difficult to read so I made a few tweaks. See what you think.

<details>
<summary>Method</summary>

```php
	private function filter_reference_url( $url = '' ) {
		$site_url = home_url();

		/**
		 * Filters the reference site URL to use for stats.
		 *
		 * This can be used to override the current site URL, for example when using the plugin on a non-public site,
		 * such as in a staging environment.
		 *
		 * @since 1.0.0
		 *
		 * @param string $site_url Reference site URL, typically the WordPress home URL.
		 */
		$reference_url = apply_filters( 'googlesitekit_site_url', $site_url );
		$reference_url = untrailingslashit( $reference_url );

		// Ensure this is not empty.
		if ( empty( $reference_url ) ) {
			$reference_url = $site_url;
		}

		// If no URL given, just return the reference site URL.
		if ( empty( $url ) ) {
			return $reference_url;
		}

		// Replace site URL with the reference site URL.
		if ( $reference_url !== $site_url ) {
			$url = str_replace( $site_url, $reference_url, $url );
		}

		return $url;
	}
```
</details>


<details>
<summary>Diff</summary>

```diff
diff --git a/includes/Context.php b/includes/Context.php
index 446a5509..57ecfc0e 100644
--- a/includes/Context.php
+++ b/includes/Context.php
@@ -368,8 +368,7 @@ final class Context {
 	 * @return string URL that starts with the reference site URL.
 	 */
 	private function filter_reference_url( $url = '' ) {
-		$orig_site_url = untrailingslashit( home_url() );
-		$site_url      = $orig_site_url;
+		$site_url = home_url();
 
 		/**
 		 * Filters the reference site URL to use for stats.
@@ -381,23 +380,22 @@ final class Context {
 		 *
 		 * @param string $site_url Reference site URL, typically the WordPress home URL.
 		 */
-		$site_url = apply_filters( 'googlesitekit_site_url', $site_url );
+		$reference_url = apply_filters( 'googlesitekit_site_url', $site_url );
+		$reference_url = untrailingslashit( $reference_url );
 
 		// Ensure this is not empty.
-		if ( empty( $site_url ) ) {
-			$site_url = $orig_site_url;
-		} else {
-			$site_url = untrailingslashit( $site_url );
+		if ( empty( $reference_url ) ) {
+			$reference_url = $site_url;
 		}
 
 		// If no URL given, just return the reference site URL.
 		if ( empty( $url ) ) {
-			return $site_url;
+			return $reference_url;
 		}
 
-		// Replace original site URL with the reference site URL.
-		if ( $orig_site_url !== $site_url ) {
-			$url = str_replace( $orig_site_url, $site_url, $url );
+		// Replace site URL with the reference site URL.
+		if ( $reference_url !== $site_url ) {
+			$url = str_replace( $site_url, $reference_url, $url );
 		}
 
 		return $url;
```
</details>

Reply:
That looks good to me, I'd just want to keep the strict `untrailingslash`ing in place.